salt/metalk8s/orchestrate/upgrade/init.sls

```diff
@@ -41,6 +41,13 @@ Skip node {{ node }}, already in {{ node_version }} newer than {{ dest_version }
 
   {%- else %}
 
+Wait for API server to be available:
+  http.wait_for_successful_query:
+  - name: https://{{ pillar.metalk8s.api_server.host }}:6443/healthz
+  - match: 'ok'
+  - status: 200
+  - verify_ssl: false
```

Comment:
I think you should move the requires here
```
      - salt: Upgrade etcd cluster
    {%- if previous_node is defined %}
      - salt: Deploy node {{ previous_node }}
    {%- endif %}
```

Reply:
Done.